store/store.go

```diff
@@ -556,6 +557,19 @@ func (s *Store) refreshDeviceSession(device *auth.DeviceState) error {
 		return fmt.Errorf("internal error: no device and auth context")
 	}
 
+	s.sessionMu.Lock()
+	defer s.sessionMu.Unlock()
+	// check that no other goroutine has already got a new session etc...
+	device1, err := s.dauthCtx.Device()
```

Comment:
Perhaps `otherDevice`? The name `device1` seems odd and easy to confuse with `device`